BrainPortal/app/controllers/demands_controller.rb

```diff
@@ -0,0 +1,431 @@
+    if params[:commit] =~ /Approve/
+      return approve_multi
+    end
+
+    if params[:commit] =~ /Fix Login/
+      return fix_login_multi
+    end
+
+    if params[:commit] =~ /Resend/
+      return resend_conf_multi
+    end
+
+    if params[:commit] =~ /Delete/
+      return delete_multi
+    end
+
+    redirect_to :action => :index
+  end
+
+
+
+  def delete_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    count = 0
+    reqs.each do |req|
+      count += 1 if req.destroy
+    end
+
+    flash[:notice] = "Deleted " + view_pluralize(count, "record") + "."
+
+    redirect_to :action => :index
+  end
+
+
+
+  def fix_login_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    @results = reqs.map do |req|
+
+      old   = req.login
+      new   = ""
+      puts "Fixing: #{old}"
+
+      email = req.email
+      if email =~ /^(\w+)@/
+        new = Regexp.last_match[1].downcase
+      end
+      new = "" if new.size < 3 || new.size > 8
+
+      if new.blank?
+        new = (req.first[0,1] + req.last[0,7]).downcase
+      end
+      new = "" if new !~ /^[a-z][a-zA-Z0-9]+$/
+      new = "" if new.size < 3 || new.size > 8
+
+      if new.blank?
+          puts "  -> No changes"
+        [ req, :no_change, 'No changes', nil ]
+      else
+        backtrace = nil
+        begin
+          req.update_attribute(:login, new)
+          puts "  -> #{old} => #{new}"
+        rescue => ex
+          backtrace = ex.backtrace
+        end
+        message = backtrace ? "Attempted" : "Adjusted"
+        [ req, :adjusted, "#{message}: #{old} => #{new}", backtrace ]
+      end
+
+    end
+
+    @results.compact!
+
+    render :action => :multi_action
+  end
+
+
+
+  def resend_conf_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    count = 0
+
+    @results = reqs.map do |req|
+      next if req.confirmed? || req.approved_by.present?
+      print "Resending confirmation: #{req.full_name}"
+      if send_confirm_email(req)
+        count += 1
+        [ req, :all_ok, "Resent confirmation email", nil ]
+      else
+        puts "=> ********* FAILED *********"
+        [ req, :failed_confirm, "ERROR: Could not send confirmation email", nil ]
+      end
+    end
+
+    @results.compact!
+
+    flash[:notice] = "Sent " + view_pluralize(count, "confirmation email") + "."
+    render :action => :multi_action
+  end
+
+
+
+  def approve_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    @results = reqs.map do |req|
+      print "Approving: #{req.full_name} => "
+      symbolic_result, message, backtrace = approve_one(req)
+
+      #message = "Account created and user notified" if message.blank? && symbolic_result == :all_ok
+      [ req, symbolic_result, message, backtrace ]
+    end
+
+    @results.compact!
+
+    render :action => :multi_action
+  end
+
+
+
+  def approve_one(req) #:nodoc:
```

Comment:
When an error occurs, I see the messages shown twice:

![screen shot 2016-09-28 at 18 08 35](https://cloud.githubusercontent.com/assets/777588/18934259/b3720296-85a6-11e6-94e2-0f42883165a3.png)